environment/lib/near2eth-watchdog.js

```diff
@@ -0,0 +1,70 @@
+const Web3 = require('web3');
+const fs = require('fs');
+const { RainbowConfig } = require('./config');
+
+function sleep (ms) {
+    return new Promise((resolve) => {
+        setTimeout(resolve, ms);
+    });
+}
+
+class Near2EthWatchdog {
+    async initialize() {
+        // @ts-ignore
+        this.web3 = new Web3(RainbowConfig.getParam('eth-node-url'));
+        let ethMasterAccount =
+            this.web3.eth.accounts.privateKeyToAccount(RainbowConfig.getParam('eth-master-sk'));
+        this.web3.eth.accounts.wallet.add(ethMasterAccount);
+        this.web3.eth.defaultAccount = ethMasterAccount.address;
+        this.ethMasterAccount = ethMasterAccount.address;
+
+        // Initialize client contract.
+        console.log('Deploying Near2EthClient contract.');
+        this.clientContract = new this.web3.eth.Contract(
+            // @ts-ignore
+            JSON.parse(fs.readFileSync(RainbowConfig.getParam('near2eth-client-abi-path'))),
+            RainbowConfig.getParam('near2eth-client-address'), {
+                from: this.ethMasterAccount,
+                handleRevert: true,
+            },
+        );
+    }
+
+    async run() {
+        while (true) {
+            const lastClientBlock = await this.clientContract.methods.last().call();
+            const latestBlock = await this.web3.eth.getBlock('latest');
```

Comment:
This probably need in try catch, otherwise it will exit from while true when exception.